django/contrib/auth/decorators.py

```diff
@@ -50,6 +50,17 @@ def login_required(function=None, redirect_field_name=REDIRECT_FIELD_NAME, login
     return actual_decorator
 
 
+def login_not_required(view_func):
+    """
+    Decorator for views that marks that the view is accessible by
+    unauthenticated users.
+    """
+    def wrapped_view(*args, **kwargs):
+        return view_func(*args, **kwargs)
+    wrapped_view.login_required = False
```

Comment:
Can you explain the reason to avoid assigning `login_required` to the passed `view_func` which would avoid the need to wrap the function?

Potential suggestion:

```py
def login_not_required(view_func):
    view_func.login_required = False
    return view_func
```